### PR TITLE
fix(jdbc): uniform exception handling

### LINKED ISSUE #7799: During update statement: terminate called without an active exception
### What happens?

I wish I could give a clear description of what exactly the bug is. But during the running of a SQL update statement, my process stops with an low level error message ("terminate called without an active exception").

This reproduces with duckdb_jdbc-0.9.0-20230531.030328-40.jar from sonatype.

### To Reproduce

Sorry for the amount of work here in the reproducer I just have not gotten it in an easy to debug state yet, as the bug is triggered by a code base that is very much just a proof of concept at this point.

On machine with java-17+, maven 3.6+, xz command line utilities and at least 36gb of RAM free, as well as about 20gb of free disk space
```
git clone https://github.com/JervenBolleman/r2s2.git
git checkout duckdb-gh-7799
mvn package
wget 'ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/rdf/uniparc_p1.rdf.xz'
mkdir test
echo -e "$(pwd)/uniparc_p1.rdf.xz\thttps://sparql.uniprot.org/uniparc" > load.txt
java -jar -Xmx4g target/r2s2-0.0.1-SNAPSHOT-uber.jar load.txt $(pwd)/test-duckdb 1
# this '1' at the end is important as the issue actually happens at step two but 
# this should get you a database in a state that allows for further debugging.
# without java or my code at all.
# running it with '0' instead should get you to the bug state in one go.
```
At then end of "step 1" in my project r2s2 you will have given the above input a 17gb duckdb.

At this of having "finished step 1 point" the following queries should trigger the exception.
```sql
UPDATE _15_iri_literal_xsd_int SET subject_13_parts= SUBSTRING(subject_13_parts,0,14)
```

The hardware I test this on is a relatively under powered machine:
```
AMD Ryzen 3 3200G 
64GB of ram
duckdb data files on zfs lz4 compressed volume on 4 slow spinning rust disks.
```

### OS:

ubuntu 23.04

### DuckDB Version:

0.8.0

### DuckDB Client:

Java on (openjdk 20.0.1)

### Full Name:

Jerven Bolleman

### Affiliation:

SIB Swiss Institute of Bioinformatics

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Mause**
Rather than adding exception handling on a method-by-method case, I added a simple wrapper generator

Fixes https://github.com/duckdb/duckdb/issues/7799

### CODE REVIEW

**Comment by hannes** [comment]
The Python script feels somewhat hacky to me, maybe there is a way to do this with templating instead?

**Comment by Mause** [comment]
> The Python script feels somewhat hacky to me, maybe there is a way to do this with templating instead?

Can you point me to what flavor of templating you'd see being usable for this? We do already use Python scripts to generate code in other places after all

**Comment by Mytherin** [comment]
Thanks! LGTM. I think templating won't work here sadly since we need to use the special JDBC function names, which unfortunately can't be generated by templates. We could try with defines but that's going to be uglier in my opinion.

File: tools/jdbc/src/jni/functions.cpp [hunk 1]
```diff
@@ -0,0 +1,353 @@
+// This file was generated by tools/jdbc/generator.py
```
Comment (felipecrv) [inline tools/jdbc/src/jni/functions.cpp:1]: What is the `h2w` invocation that triggers this? @Mause

Comment (Mause) [reply tools/jdbc/src/jni/functions.cpp:1]:
```sh
python3 generator.py
```
?

Comment (felipecrv) [reply tools/jdbc/src/jni/functions.cpp:1]: Hmmm. Thank you for confirming.

When I tried running it before I've got many `header2whathever` related errors and suspected the script wasn't supposed to be called directly, but it works now. I think I was calling it from the wrong path.

**Pull request merged**